wordpress-mobile/WordPress-iOS
# Add autosave attributes to Core Data

Post autosave attributes are saved to Core Data upon retrieval from the remote service.

Issue: wordpress-mobile/WordPress-iOS#12141
Related PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/192

Although the changes in the WordPressKit PR map all autosave attributes, only those necessary for wordpress-mobile/WordPress-iOS#12141 are now persisted to Core Data.

### Testing Details

1. On WordPress.com, create a new post or open an existing post for editing
2. Set the title to “Hello, world!”, the body to “One.”, and the excerpt to "An excerpt"
3. Publish the post
4. Re-open the post and add a new line, “Two.” – then close the browser tab without saving
5. Again, re-open the post and notice the following message -  “The backup of this post in your browser is different from the version below. Restore the backup”  – this signifies that an autosave was made
6. Run the iOS app from Xcode and [locate the path to iOS app Core Data's sqlite file](https://stackoverflow.com/a/47402447/1305067), `WordPress.sqlite` and open a database browser at this location
6. On the WordPress for iOS app, under the My Sites tab, navigate to the list of posts containing the post
8. Run the query `SELECT ZAUTOSAVETITLE, ZAUTOSAVEEXCERPT, ZAUTOSAVECONTENT, ZAUTOSAVEMODIFIEDDATE FROM ZBASEPOST;` and observe that the title, excerpt, content, and timestamp contain the expected values:

<img width="1028" alt="Screen Shot 2019-11-01 at 12 23 02 AM" src="https://user-images.githubusercontent.com/1898325/68000596-80780800-fc3f-11e9-8073-c22b7840cf21.png">
